src/qir/boolcodegenextension_workaround.rs

```diff
@@ -8,21 +8,25 @@ use hugr_llvm::inkwell::IntPredicate;
 use hugr_llvm::inkwell::types::IntType;
 use hugr_llvm::sum::LLVMSumValue;
 use hugr_llvm::types::TypingSession;
-use tket2::extension::bool::{BOOL_EXTENSION_ID, BoolOp, ConstBool};
+use tket::extension::bool::{BOOL_EXTENSION_ID, BoolOp, ConstBool};
 
 // This is copied from https://github.com/CQCL/guppylang/blob/main/execute_llvm/src/bool.rs
 // And is a workaround until the below issues are resolved
 // https://github.com/CQCL/tket2/issues/909
 // https://github.com/CQCL/tket2/issues/910
 
+#[allow(dead_code)]
 const BOOL_TYPE_ID: TypeName = TypeName::new_inline("bool");
 
+#[allow(dead_code)]
 fn llvm_bool_type<'c>(ts: &TypingSession<'c, '_>) -> IntType<'c> {
     ts.iw_context().bool_type()
 }
+#[allow(dead_code)]
 pub struct BoolCodegenExtension;
 
 impl BoolCodegenExtension {
+    #[allow(dead_code)]
```

Comment:
`expect` will emit an error when the skip is no longer needed, so we know to take it away.

Why do we have dead code?


```suggestion
#[expect(dead_code)]
const BOOL_TYPE_ID: TypeName = TypeName::new_inline("bool");

#[expect(dead_code)]
fn llvm_bool_type<'c>(ts: &TypingSession<'c, '_>) -> IntType<'c> {
ts.iw_context().bool_type()
}
#[expect(dead_code)]
pub struct BoolCodegenExtension;

impl BoolCodegenExtension {
    #[expect(dead_code)]
```

Reply:
I am not really sure why the error shows up now, before the last two commits the error from clippy did not show up. If I try to remove it, the build fails.

Reply:
Ah, I guess it's because it's a private definition only used for testing (but always compiled).

I say just keep it the `allow`, or mark the `mod boolcodegenextension_workaround;` in `qir.rs` with `#[cfg(test)]`.